File: AnimalFostering.API/Models/AlertRule.cs
using System.ComponentModel.DataAnnotations;

namespace AnimalFostering.API.Models
{
    public class AlertRule
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Metric { get; set; } = "PendingApplications";

        [Required]
        public string Comparison { get; set; } = "GreaterThan";

        [Required]
        public int Threshold { get; set; }

        public bool IsActive { get; set; } = true;

        public int CreatedByUserId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public User CreatedByUser { get; set; } = null!;
    }
}

File: animal_fostering_app/lib/models/alert_rule.dart
class AlertRule {
  final int id;
  final String name;
  final String metric;
  final String comparison;
  final int threshold;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  AlertRule({
    required this.id,
    required this.name,
    required this.metric,
    required this.comparison,
    required this.threshold,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory AlertRule.fromJson(Map<String, dynamic> json) {
    return AlertRule(
      id: json['id'] as int,
      name: json['name'] as String,
      metric: json['metric'] as String,
      comparison: json['comparison'] as String,
      threshold: json['threshold'] as int,
      isActive: json['isActive'] as bool,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
}
